### DISCUSSION TITLE
Understanding Docker build system

### DISCUSSION THREAD

**Discussion by benjcunningham** [milvus-io/milvus#4822] [Q&A and General discussion] [answered]
Hello! I've noticed that the main [milvusdb/milvus repo on Docker Hub](https://hub.docker.com/r/milvusdb/milvus/tags?page=1&ordering=last_updated) only has images built on CentOS. However, I've also noticed these things:

- In [the documentation](https://www.milvus.io/docs/milvus_docker-cpu.md) there is mention of Ubuntu LTS <sup>1</sup>
- In [milvusdb/daily-build](https://hub.docker.com/r/milvusdb/daily-build/tags?page=1&ordering=last_updated&name=ubuntu) there are Ubuntu daily builds for versions up to 0.10.5
- I've seen Ubuntu listed in the build matrix and some Dockerfiles in the repo (though to be honest, I haven't fully understood how the build/release process works yet)

So is there a reason there aren't Ubuntu images in milvusdb/milvus for official releases? I'm just curious to know, so I can understand the project a bit better. 😄 

---

<sup>1</sup> Though this is a bit confusing, because it's listed as an OS "requirement"

**Comment by yhmo** [comment, marked as answer]
We develop milvus under Unbuntu 18, but the release docker image is make from CentOS. The main reason is: 
1. CentOS is more popular in enterprise applications
2. From our testing experience, the openblas(a parallel compute library used by faiss) run a bit fast under CentOS than Ubuntu.
